packages/framework/src/hooks/useRegisterBindings.ts

```diff
@@ -88,6 +88,10 @@ export const useRegisterBindings = <T extends { [key: string]: unknown }>(
     }
   }, [options?.forceState]);
 
+  useEffect(() => {
+    setWidgetState({});
+  }, [setWidgetState]);
```

Comment:
@evshi this basically reset the values of that widget on widget unmounted. and thus on next render of that element will re-render and again that in `widgetState`